### ISSUE TITLE
Need to expose image and kubelet path overrides via CLI flags

### ISSUE THREAD

**Issue by brandond** [rancher/rke2#542]
The RKE2 core images can be overridden by undocumented environment variables. The kubelet cannot currently be overridden - RKE2 always uses the one from the runtime image.

In order to swap in components from other distributions, we need to allow for these to be overridden via actual supported CLI flags.

**Comment by brandond** [comment]
This was fixed in the linked issue.

**Comment by davidnuzik** [comment]
This is merged in master but is in 1.21.1 milestone. We haven't released 1.21.0 yet. Moving to 1.21.0 milestone.

**Comment by ShylajaDevadiga** [comment]
Validated  experimental capabilities of leveraging components of users choice. Created RKE2 cluster with Amazon EKS Distro components after generating the RKE2 configs for use with Amazon EKS Distro 

```
journalctl -u rke2-server |grep eks
May 12 13:48:17 ip-172-31-40-125 rke2[21951]: time="2021-05-12T13:48:17Z" level=info msg="Pulling image public.ecr.aws/eks-distro/etcd-io/etcd:v3.4.14-eks-1-18-1..."
May 12 13:48:30 ip-172-31-40-125 rke2[21951]: time="2021-05-12T13:48:30Z" level=info msg="Pulling image public.ecr.aws/eks-distro/kubernetes/kube-apiserver:v1.18.9-eks-1-18-1..."
May 12 13:48:34 ip-172-31-40-125 rke2[21951]: time="2021-05-12T13:48:34Z" level=info msg="Pulling image public.ecr.aws/eks-distro/kubernetes/kube-controller-manager:v1.18.9-eks-1-18-1..."
May 12 13:48:38 ip-172-31-40-125 rke2[21951]: time="2021-05-12T13:48:38Z" level=info msg="Pulling image public.ecr.aws/eks-distro/kubernetes/kube-scheduler:v1.18.9-eks-1-18-1..."

```
```
root@ip-172-31-40-125:~# sudo /var/lib/rancher/rke2/bin/crictl -r unix:///run/k3s/containerd/containerd.sock images
IMAGE                                                          TAG                              IMAGE ID            SIZE
docker.io/rancher/hardened-calico                              v3.13.3-build20210223            736cae9d947ba       158MB
docker.io/rancher/hardened-flannel                             v0.13.0-rancher1-build20210223   366c64051af85       92.3MB
docker.io/rancher/klipper-helm                                 v0.4.3-build20210225             e7c578d94f6d9       50.7MB
docker.io/rancher/nginx-ingress-controller-defaultbackend      1.5-rancher1                     b5af743e59849       2MB
docker.io/rancher/nginx-ingress-controller                     nginx-0.30.0-rancher1            2166188b0dba1       114MB
public.ecr.aws/eks-distro/coredns/coredns                      v1.7.0-eks-1-18-1                6dbf7f0180dba       14.2MB
public.ecr.aws/eks-distro/etcd-io/etcd                         v3.4.14-eks-1-18-1               e77eead05c5e3       203MB
public.ecr.aws/eks-distro/kubernetes-sigs/metrics-server       v0.4.0-eks-1-18-1                7c79867073e0e       202MB
public.ecr.aws/eks-distro/kubernetes/kube-apiserver            v1.18.9-eks-1-18-1               a2ea61c746e10       220MB
public.ecr.aws/eks-distro/kubernetes/kube-controller-manager   v1.18.9-eks-1-18-1               b50f3c224c59b       218MB
public.ecr.aws/eks-distro/kubernetes/kube-proxy                v1.18.9-eks-1-18-1               7b3d7533dd469       231MB
public.ecr.aws/eks-distro/kubernetes/kube-scheduler            v1.18.9-eks-1-18-1               3f6c60b314759       203MB
public.ecr.aws/eks-distro/kubernetes/pause                     v1.18.9-eks-1-18-1               ff45cda5b28a9       307kB
root@ip-172-31-40-125:~# 
```

```
kubectl version
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.6+rke2r1", GitCommit:"8a62859e515889f07e3e3be6a1080413f17cf2c3", GitTreeState:"clean", BuildDate:"2021-04-21T20:09:25Z", GoVersion:"go1.15.8b5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.9-eks-1-18-1", GitCommit:"94f372e501c973a7fa9eb40ec9ebd2fe7ca69848", GitTreeState:"archive", BuildDate:"2020-09-16T13:46:48Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```
```
kubectl get nodes
NAME               STATUS   ROLES                       AGE    VERSION
ip-172-31-40-125   Ready    control-plane,etcd,master   9m3s   v1.18.9-eks-1-18-1
```
```
kubectl get pods -A
NAMESPACE     NAME                                                  READY   STATUS      RESTARTS   AGE
kube-system   etcd-ip-172-31-40-125                                 1/1     Running     0          23m
kube-system   helm-install-rke2-canal-c5drp                         0/1     Completed   0          24m
kube-system   helm-install-rke2-coredns-sb7r6                       0/1     Completed   0          24m
kube-system   helm-install-rke2-ingress-nginx-rssjv                 0/1     Completed   0          24m
kube-system   helm-install-rke2-kube-proxy-96t6z                    0/1     Completed   0          24m
kube-system   helm-install-rke2-metrics-server-njpxt                0/1     Completed   0          24m
kube-system   kube-apiserver-ip-172-31-40-125                       1/1     Running     0          23m
kube-system   kube-controller-manager-ip-172-31-40-125              1/1     Running     0          24m
kube-system   kube-proxy-2866v                                      1/1     Running     0          24m
kube-system   kube-scheduler-ip-172-31-40-125                       1/1     Running     0          24m
kube-system   rke2-canal-srn86                                      2/2     Running     0          24m
kube-system   rke2-coredns-rke2-coredns-6985c7676c-gmjt2            1/1     Running     0          24m
kube-system   rke2-ingress-nginx-controller-d4989f458-6kphx         1/1     Running     0          23m
kube-system   rke2-ingress-nginx-default-backend-65f75d6664-cv9rt   1/1     Running     0          23m
kube-system   rke2-metrics-server-7665f9fb8b-wl6fd                  1/1     Running     0          23m
```
